modules/tinymce/src/core/demo/ts/demo/FullDemo.ts

```diff
@@ -1,12 +1,55 @@
 /* eslint-disable no-console */
-import { Merger } from '@ephox/katamari';
+import { Cell } from '@ephox/katamari';
 import { SugarElement } from '@ephox/sugar';
 
 import { Editor, RawEditorOptions, TinyMCE } from 'tinymce/core/api/PublicApi';
 
 declare let tinymce: TinyMCE;
 
 export default (): void => {
+  const buildModes = (ed: Editor) => {
+    const selectedMode = Cell<string>(ed.mode.get());
+    const setMode = (mode: string) => {
+      ed.mode.set(mode);
+      selectedMode.set(mode);
+    };
+    const makeModeUI = (mode: string) => {
+      const toggleSpec = {
+        text: `${mode} Mode`,
+        onAction: () => {
+          console.log('Current mode: ', ed.mode.get());
+          ed.mode.get() === mode ? ed.mode.set(selectedMode.get()) : setMode(mode);
+          console.log('New mode: ', ed.mode.get());
+        },
+        onSetup: (api: any) => {
+          const toggleActive = () => {
+            api.setActive(ed.mode.get() === mode);
+            api.setEnabled(true);
+          };
+          toggleActive();
+          ed.on('SwitchMode', toggleActive);
+          return () => ed.off('SwitchMode', toggleActive);
+        }
```

Comment:
Not even sure any of this should carry over. It served it's purpose

Reply:
I think it's useful to have in a demo somewhere, but a lot simpler. Perhaps a new demo just focused on different types of things that respond to these changes.

Reply:
Moved these changes to a separate demo. Changes in FullDemo are reverted.